test/integration/examples/apiserver_test.go

```diff
@@ -225,164 +232,143 @@ func TestAPIServiceWaitOnStart(t *testing.T) {
 	}
 }
 
-func TestAggregatedAPIServer(t *testing.T) {
+type roundTripperFunc func(*http.Request) (*http.Response, error)
+
+func (f roundTripperFunc) RoundTrip(req *http.Request) (*http.Response, error) {
+	return f(req)
+}
+
+// TestFrontProxyConfig tests that the RequestHeader configuration is consumed
+// correctly by the aggregated API servers.
+func TestFrontProxyConfig(t *testing.T) {
```

Comment:
We also need a test to show that KAS itself can consume the UID header config (i.e. it should not consume UIDs unless the new flag is set).  I think the `certreload` integration tests may have most of the logic that you need.

Reply:
added 2 integration tests for cases where remote UID headers are set and when they are not